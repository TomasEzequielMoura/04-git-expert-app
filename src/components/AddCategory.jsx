import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({ onNewCategory }) => {

    const [inputValue, setInputValue] = useState('')


    const onInputChange = (event) => {
        setInputValue(event.target.value)
    }

    const onSubmit = (event) => {
        event.preventDefault();
        if (inputValue.trim().length <= 1);
        else {
            onNewCategory(inputValue.trim())
            setInputValue('')
        }
    }

    return (
        <>
            <form onSubmit={onSubmit} aria-label='form'>
                <input
                    type='text'
                    placeholder='Buscar Gifs'
                    value={inputValue}
                    onChange={(event) => onInputChange(event)}
                />
            </form>

        </>
    )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired,
}