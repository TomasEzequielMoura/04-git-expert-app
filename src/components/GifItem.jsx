import React from 'react'

const GifItem = ({title, url, id}) => {

    return (
        <>
            <div className='card'>
                <img src={url}/>
                <p>{title}</p>
            </div>
        </>
    )
}

export default GifItem