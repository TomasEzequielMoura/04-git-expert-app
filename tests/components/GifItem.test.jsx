import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem";

describe('Pruebas en <GifItem />', () => {

    const alt = 'Hola';
    const title = 'Hola';
    const url = 'https://hola.com.ar/';

    test('debe de hacer match con el snapshot', () => {
        
        const {container} = render(<GifItem title={alt} url={url}/>); 
        expect(container).toMatchSnapshot();
    }) 

    test('debe de mostrar la imagen con el URL y el ALT indicado', () => {

        render(<GifItem title={title} url={url}/>);
        // console.log(screen.getByRole('img').alt) 

        // forma basica
        // expect(screen.getByRole('img').src).toBe(url);
        // expect(screen.getByRole('img').alt).toBe(title);

        // desestructuramos
        const {src, alt} = screen.getByRole('img')
        expect(src).toBe(url);
        expect(alt).toBe(alt);

    }) 

    test('debe de mostrar el titulo en el componente', () => {
        render(<GifItem title={alt} url={url}/>)  

        expect(screen.getByText(title)).toBeTruthy()
        // expect(screen.getByText(title).innerHTML).toContain(title)
    }) 
})