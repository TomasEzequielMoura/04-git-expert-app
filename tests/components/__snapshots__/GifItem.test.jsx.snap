// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Hola"
      src="https://hola.com.ar/"
    />
    <p>
      Hola
    </p>
  </div>
</div>
`;
